### DISCUSSION TITLE
[Question][Hw1] How to distinguish between an integer constant and a boolean constant

### DISCUSSION THREAD

**Discussion by frogfrog2** [prosyslab-classroom/cs424-program-reasoning#286] [Q&A]
Name: Jihwan Park

Within the function eval, Llvm.classify_value distinguishes whether 'e' is a constant or a variable. It seems that constants are still divided into integer constants and boolean constants, but I have no idea how to distinguish between them. I tried to use Llvm.type_of function, but I couldn't find how lltypes corresponding to integer and boolean are expressed in Ocaml. (maybe something similar to i1 and i8) Any advice on the above issue would be greatly appreciated!

**Comment by goodtaeeun** [comment]
The `eval` function will not encounter boolean constants. This is because boolean values are only generated from comparisons. Thus, in the memory model of our interpreter, they don't exist as constants, but as `Value.Bool` types.
